plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessPlugin.java

```diff
@@ -36,7 +36,14 @@ public void apply(Project project) {
 
 		// clear spotless' cache when the user does a clean
 		Task clean = project.getTasks().getByName(BasePlugin.CLEAN_TASK_NAME);
-		clean.doLast(unused -> SpotlessCache.clear());
+		clean.doLast(unused -> {
+			// resolution for: https://github.com/diffplug/spotless/issues/243#issuecomment-564323856
+			// project.getRootProject() is consistent across every project, so only of one the clears will
+			// actually happen (as desired)
+			//
+			// we use System.identityHashCode() to avoid a memory leak by hanging on to the reference directly
+			SpotlessCache.clearOnce(System.identityHashCode(project.getRootProject()));
```

Comment:
@nedtwigg Reading the source code for `SpotlessCache::clearOnce`, I'm unclear on why passing `project.getRootProject()` to the method would cause the method to hang onto it; AFAICT the method never stores the reference inside a field or something similar. :thinking: 

So I was wondering if you could help me understand why the call to `System::identityHashCode` is needed here?

Reply:
https://github.com/diffplug/spotless/blob/f73660f96cfcd3bbc820f0bc4889a4dd74fa060d/lib/src/main/java/com/diffplug/spotless/SpotlessCache.java#L113

Reply:
This might be a better way to spell it, but I think the javadoc is the clearest explanation.

```java
if (key == null || !key.equals(lastClear)) {
	lastClear = key;
} else {
	// only clear once
	return false;
}
```

Reply:
Aha, that `lastClear = key` line is what I'd missed! Thank you @nedtwigg. :)